modules/stochastic_tools/src/surrogates/ActiveLearningGaussianProcess.C

```diff
@@ -0,0 +1,116 @@
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ActiveLearningGaussianProcess.h"
+
+#include <petsctao.h>
+#include <petscdmda.h>
+
+#include "libmesh/petsc_vector.h"
+#include "libmesh/petsc_matrix.h"
+
+#include <math.h>
+
+registerMooseObject("StochasticToolsApp", ActiveLearningGaussianProcess);
+
+InputParameters
+ActiveLearningGaussianProcess::validParams()
+{
+  InputParameters params = SurrogateTrainerBase::validParams();
+  params.addClassDescription(
+      "Permit re-training Gaussian Process surrogate model for active learning.");
+  params.addRequiredParam<UserObjectName>("covariance_function", "Name of covariance function.");
+  params.addParam<bool>(
+      "standardize_params", true, "Standardize (center and scale) training parameters (x values)");
+  params.addParam<bool>(
+      "standardize_data", true, "Standardize (center and scale) training data (y values)");
+  MooseEnum tuning_type("tao adam none", "none");
+  params.addRequiredParam<MooseEnum>(
+      "tuning_algorithm", tuning_type, "Hyper parameter optimizaton algorithm");
+  params.addParam<unsigned int>("iter_adam", 1000, "Tolerance value for Adam optimization");
+  params.addParam<unsigned int>("batch_size", "The batch size for Adam optimization");
+  params.addParam<Real>("learning_rate_adam", 0.001, "The learning rate for Adam optimization");
+  params.addParam<std::string>(
+      "tao_options", "", "Command line options for PETSc/TAO hyperparameter optimization");
+  params.addParam<bool>(
+      "show_optimization_details", false, "Switch to show TAO or Adam solver results");
+  params.addParam<std::vector<std::string>>("tune_parameters",
+                                            "Select hyperparameters to be tuned");
+  params.addParam<std::vector<Real>>("tuning_min", "Minimum allowable tuning value");
+  params.addParam<std::vector<Real>>("tuning_max", "Maximum allowable tuning value");
+  return params;
+}
+
+ActiveLearningGaussianProcess::ActiveLearningGaussianProcess(const InputParameters & parameters)
+  : SurrogateTrainerBase(parameters),
+    CovarianceInterface(parameters),
+    SurrogateModelInterface(this),
+    _gp_handler(declareModelData<StochasticTools::GaussianProcessHandler>("_gp_handler")),
+    _training_params(declareModelData<RealEigenMatrix>("_training_params")),
+    _standardize_params(getParam<bool>("standardize_params")),
+    _standardize_data(getParam<bool>("standardize_data")),
+    _optimization_opts(StochasticTools::GaussianProcessHandler::GPOptimizerOptions(
+        getParam<MooseEnum>("tuning_algorithm"),
+        getParam<std::string>("tao_options"),
+        getParam<bool>("show_optimization_details"),
+        getParam<unsigned int>("iter_adam"),
+        isParamValid("batch_size") ? getParam<unsigned int>("batch_size") : 0,
```

Comment:
Make the `"batch_size"` parameter default to 0. And have the error below be `if (getParam<unsigned int>("batch_size") > 0 && _optimization_opts.opt_type == "tao")